### PR TITLE
Remove legacy C code

### PR DESCRIPTION
**Pull request by TheOneRing**

### CODE REVIEW

**Comment by guruz** [comment]
I'm quite sure there was a reason behind this :-)

Let's not merge before @ogoffart confirms

**Comment by TheOneRing** [comment]
The last remaning stuff can probably only be killed once we can use std::path wich requires c++17 and mac 10.15

File: test/csync/vio_tests/check_vio_ext.cpp [hunk 1]
```diff
@@ -125,8 +101,7 @@ static int teardown(void **state) {
     rc = wipe_testdir();
     assert_int_equal(rc, 0);
 
-    SAFE_FREE(((statevar*)*state)->result);
-    SAFE_FREE(*state);
+    delete reinterpret_cast<statevar*>(*state);
```
Comment (dschmidt) [inline test/csync/vio_tests/check_vio_ext.cpp:104]: result doesn't need to be freed anymore?

Comment (TheOneRing) [reply test/csync/vio_tests/check_vio_ext.cpp:104]: I now use new and delete instead of malloc

**Pull request merged**